### PR TITLE
cdecker's weekly cleanups

### PR DESCRIPTION
**Pull request by cdecker**
Things that don't really fit in anywhere else:

 - Avoid trying to restart the `bitcoind` proxy along with `lightingd`, removes the `unable to bind` warning
 - A small fix to wait for a connection to be recognized before trying to use it
 - Marking some flaky tests as `@flaky` (will do a stabilization pass later, but they were really annoying and were addressed only by restarting instead of fixing them).
 - Remove the `UPDATE <message_type>` spam from `subd` which is really annoying in production.

### CODE REVIEW

**Comment by cdecker** [comment]
Just found another simple to fix test, so I just added it:

 - upfront shutdown script was reconnecting manually but disallowing reconnections...

File: lightningd/subd.c [hunk 1]
```diff
@@ -469,8 +469,6 @@ static struct io_plan *sd_msg_read(struct io_conn *conn, struct subd *sd)
 		}
 	}
 
-	log_debug(sd->log, "UPDATE %s", sd->msgname(type));
```
Comment (niftynei) [inline lightningd/subd.c]: would it be more appropriate to move this to log at the IO level?

Comment (cdecker) [reply lightningd/subd.c]: The IO level logs only raw messages (hex encoded of course) so far afaik

Comment (rustyrussell) [reply lightningd/subd.c]: It *can* log messages, but we logged this in lieu of full IO.  We should probably log internal IO messages, but that's a separate change.

**Comment by rustyrussell** [comment]
> Just found another simple to fix test, so I just added it:
> 
>     * upfront shutdown script was reconnecting manually but disallowing reconnections...

Can't reproduce this here.  may_reconnect=False does two things: disable automatic reconnect, and check that we don't get reconnections manually.

A "reconnection" is only when we have a live channel, and this test does check l1 is onchain. Easiest fix is to change that to check l2 is onchain as well.

File: lightningd/subd.c (hunk 1 above)
Comment (trueptolemy) [reply lightningd/subd.c]: Does the _internal IO messages_ mean the message between process?

**Comment by cdecker** [comment]
> > Just found another simple to fix test, so I just added it:
> > ```
> > * upfront shutdown script was reconnecting manually but disallowing reconnections...
> > ```
> 
> Can't reproduce this here. may_reconnect=False does two things: disable automatic reconnect, and check that we don't get reconnections manually.
> 
> A "reconnection" is only when we have a live channel, and this test does check l1 is onchain. Easiest fix is to change that to check l2 is onchain as well.

Decided to drop this commit, in order not to hold up. Will resubmit a better, less driveby, fix in the next round :-)

**Pull request merged**